### PR TITLE
Merging demo from NAWI Spring Meeting Pt2 - NF+Bypass example

### PR DESCRIPTION
**Pull request by adam-a-a**
## Fixes/Resolves:
#1061 
## Summary/Motivation:
Handle NF+Bypass test failures and resolve in this PR.

## Changes proposed in this PR:
- add NF with bypass example

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
This needs to have the Linux/Win failures sorted out before being able to merge it. @lbianchi-lbl, @adam-a-a, @avdudchenko, and @bknueven will sync offline to see if there's any way to speed up iterating on this since the Linux failures cannot be reproduced on WSL.

**Comment by adam-a-a** [comment]
@avdudchenko @lbianchi-lbl @bknueven @dangunter @MichaelPesce - I managed to get all checks to pass, as finnicky as these flowsheets were (and probably still are)! I just need to skip the mac test. Please have a look, and hopefully, we'll be able to have the GUI segment of the spring meeting demo in place for the June release.

**Review by bknueven — APPROVED** [review]
It's clear from the Ipopt logs in the jupyter notebook that these flowsheets are not quite numerically stable.

If we want to merge this before the June release we will need to skip the macOS tests for certain.

**Comment by adam-a-a** [comment]
> these flowsheets are not quite numerically stable.

That's certainly the case. Although I was lucky enough to get windows and Linux checks to finally pass, the notebook tests were "suddenly" failing, which I didn't expect. 

I am hopeful that we can further improve these examples in a subsequent PR.

File: watertap/examples/flowsheets/nf_dspmde/nf_ui.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+from pyomo.environ import units as pyunits
+from idaes.core.solvers import get_solver
+
+
+def export_to_ui():
+    return FlowsheetInterface(
+        name="NF-DSPM-DE",
+        do_export=export_variables,
+        do_build=build_flowsheet,
+        do_solve=solve_flowsheet,
```
Comment (lbianchi-lbl) [inline watertap/examples/flowsheets/nf_dspmde/nf_ui.py:23]:
```suggestion
        do_solve=solve_flowsheet,
        requires_idaes_solver=True,
```

File: watertap/examples/flowsheets/nf_dspmde/nf_with_bypass_ui.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    units as pyunits,
+)
+
+
+def export_to_ui():
+    return FlowsheetInterface(
+        name="NF-DSPM-DE with bypass",
+        do_export=export_variables,
+        do_build=build_flowsheet,
+        do_solve=solve_flowsheet,
```
Comment (lbianchi-lbl) [inline watertap/examples/flowsheets/nf_dspmde/nf_with_bypass_ui.py:27]:
```suggestion
        do_solve=solve_flowsheet,
        requires_idaes_solver=True,
```

**Comment by lbianchi-lbl** [comment]
@adam-a-a I've left a couple of suggestions that should take care of the fsapi tests that are currently still failing.

**Comment by lbianchi-lbl** [comment]
There's still one remaining failure from the `fsapi`/`nf_dspmde.nf_ui` combo:

image

The `requires_idaes_solvers=True` kwarg seems to have worked for `nf_dspmde.nf_with_bypass_ui` (i.e. it's not failing anymore), so I wonder what the difference could be. Maybe the failure for `nf_dspmde.nf_ui` happens at an earlier stage, which prevents the fsapi object to be used at all for testing?

**Comment by lbianchi-lbl** [comment]
@adam-a-a the UI flowsheet that was still failing the Intel macOS CI check has been dealt with (by explicitly excluding it at the pytest command invocation level :man_shrugging: ).

**Review by TimBartholomew — APPROVED** [review]

**Pull request merged**